pkg/builders/s2i/builder.go

```diff
@@ -319,7 +319,7 @@ func (b *Builder) Build(ctx context.Context, f fn.Function, platforms []fn.Platf
 	}()
 
 	opts := types.ImageBuildOptions{
-		Tags:       []string{f.Build.Image},
+		Tags:       []string{f.Image},
```

Comment:
Same comment here, I presumed we'd have a check early in the function which uses `f.Image` if defined, defaulting to `f.Build.Image`, with an error if neither are defined.